File: Decorator/Java/DataServiceDecorator/src/DataServiceDecorator/LoggingDecorator.java
package DataServiceDecorator;

import java.util.logging.Logger;

public class LoggingDecorator extends DataServiceDecorator
{
    private final Logger _logger;

    public LoggingDecorator(DataServiceComponent dataService, Logger logger) {
    	super(dataService);
        _logger = logger;
    }

    @Override
    public String getText() {
        _logger.info("Start fetching text");
        var text = dataService.getText();
        _logger.info("End fetching text");
        return text;
    }
}

File: Decorator/C#/DataServiceDecorator/LoggingDecorator.cs
using DataServiceDecorator.Infrastructure;

namespace DataServiceDecorator
{
    public class LoggingDecorator : DataServiceDecorator
    {
        private readonly Logger _logger;

        public LoggingDecorator(DataServiceComponent dataService, Logger logger) : base(dataService)
        {
            _logger = logger;
        }

        public override string GetText()
        {
            _logger.Log("Start fetching text");
            var text = DataService.GetText();
            _logger.Log("End fetching text");
            return text;
        }
    }
}